tests/extension/oneapi_weak_object/weak_object_ownership.cpp

```diff
@@ -0,0 +1,41 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2023 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include "weak_object_common.h"
+
+namespace weak_object_ownership {
+using namespace weak_object_common;
+
+TEST_CASE("weak_object weak_object_ownership", "[weak_object]") {
+#ifdef SYCL_EXT_ONEAPI_WEAK_OBJECT
+  test_weak_object_ownership<sycl::accessor<int>>{}("accessor");
+  test_weak_object_ownership<sycl::host_accessor<int>>{}("host_accessor");
+  test_weak_object_ownership<sycl::buffer<int>>{}("buffer");
+  test_weak_object_ownership<sycl::context>{}("context");
+  test_weak_object_ownership<sycl::event>{}("event");
+  test_weak_object_ownership<sycl::queue>{}("queue");
+
+  test_weak_object_ownership<sycl::queue>::check_type();
+  test_weak_object_ownership<sycl::queue>{}.test_local_types();
+#else
+  SKIP("SYCL_EXT_ONEAPI_WEAK_OBJECT is not defined");
+#endif
```

Comment:
This seems to be a common idiom for all extension test cases, so it makes sense to move common part to a specialized macro to test extensions functionality, which can be additionally parameterized with `SYCL_EXT_*` macro. What do you think?

Reply:
We have to check that `SYCL_EXT_*` macro is defined for each extension, but we can't call `#ifdef`, `#else` and `#endif` macros inside `#define`, so I think it is not possible to create a single specialized a macro. At best, we can use simple `if` statement and it would work fine until extension macro is defined, but there is a compfail otherwise.

```
#define TEST_EXT_FUNCTIONALITY(EXT_MACRO, ...) {  \
  if (!EXT_MACRO) {                               \
    SKIP(#EXT_MACRO + " is not defined");         \
  } else {                                        \
    __VA_ARGS__                                   \
  }                                               \
}
```